dashboard/modules/serve/schema.py

```diff
@@ -270,6 +289,63 @@ def route_prefix_format(cls, v):
                 'contain "{" or "}".'
             )
 
+        return v
+
+
+class ServeInstanceSchema(BaseModel):
+    deployments: List[DeploymentSchema] = Field(...)
+
+
+def deployment_to_schema(d: Deployment) -> DeploymentSchema:
+    if d.ray_actor_options is not None:
+        ray_actor_options_schema = RayActorOptionsSchema.parse_obj(d.ray_actor_options)
+    else:
+        ray_actor_options_schema = None
+
+    return DeploymentSchema(
+        name=d.name,
+        import_path=d.func_or_class,
```

Comment:
hmm what will happen if this is a serialized class? we should make the change to only support import_paths in the controller so this makes sense

Reply:
Good question, that would cause an issue only when using both the REST API and the Python API at the same time. I can add handling to avoid setting the import path to a serialized function/class.

Making the controller only use import paths is a good idea, and I believe we’re making that change as a part of the controller unification. Since it’s somewhat extensive, I’d rather include it in a follow-up change.

Reply:
yes sounds good, file an issue for it?

Reply:
Sure thing, filed it [here](https://github.com/ray-project/ray/issues/22604).